dask_glm/algorithms.py

```diff
@@ -219,21 +224,47 @@ def wrapped(beta, X, y, z, u, rho):
             reltol * np.linalg.norm(rho * u)
 
         if primal_res < eps_pri and dual_res < eps_dual:
-            print("Converged!", k)
             break
 
     return z
 
 
 def local_update(X, y, beta, z, u, rho, f, fprime, solver=fmin_l_bfgs_b):
-
     beta = beta.ravel()
     u = u.ravel()
     z = z.ravel()
-    solver_args = (X, y, z, u, rho)
-    beta, f, d = solver(f, beta, fprime=fprime, args=solver_args,
-                        maxiter=200,
-                        maxfun=250)
+
+    if len(X) > 1:
+        # Construct dask graph for computation
+        X = [da.from_array(x, chunks=x.shape, name=False,
+                           getitem=operator.getitem) for x in X]
+        X = da.concatenate(X, axis=1).persist(get=dask.get)
+        beta = da.from_array(beta, chunks=beta.shape, name=False,
+                             getitem=operator.getitem).persist(get=dask.get)
+        ff = f(beta, X, y, z, u, rho)
+        ffprime = fprime(beta, X, y, z, u, rho)
+        ff, ffprime = dask.delayed(ff), dask.delayed(ffprime)
+        dsk = ensure_dict(sharedict.merge(ff.dask, ffprime.dask))
+
+        beta_key = beta._keys()[0]
+        def f2(beta):
+            """ Reuse existing graph, just swap in new beta and compute """
+            dsk[beta_key] = beta
+            result, gradient = dask.get(dsk, [ff.key, ffprime.key])
+            print(result, gradient)
+            return result, gradient
+
+        solver_args = ()
+    else:
+        X = X[0]
+        solver_args = (X, y, z, u, rho)
+
+        def f2(beta, *args):
+            result, gradient = f(beta, *args), fprime(beta, *args)
+            print(result, gradient)
+            return result, gradient
+
+    beta, _, _ = solver(f2, beta, args=solver_args, maxiter=200, maxfun=250)
```

Comment:
Oh, it looks like you're right; sorry, I've never called it that way. Hmm back to the drawing board.